openupgradelib/openupgrade.py

```diff
@@ -896,15 +896,41 @@ def rename_xmlids(cr, xmlids_spec):
 
     :param xmlids_spec: a list of tuples (old module.xmlid, new module.xmlid).
     """
+    get_data_query = (
+        """SELECT res_id, model FROM ir_model_data
+        WHERE module=%s AND name=%s""")
+
     for (old, new) in xmlids_spec:
         if '.' not in old or '.' not in new:
             logger.error(
                 'Cannot rename XMLID %s to %s: need the module '
                 'reference to be specified in the IDs' % (old, new))
-        else:
-            query = ("UPDATE ir_model_data SET module = %s, name = %s "
-                     "WHERE module = %s and name = %s")
-            logged_query(cr, query, tuple(new.split('.') + old.split('.')))
+            continue
+
+        cr.execute(get_data_query, tuple(old.split('.')))
+        old_row = cr.fetchone()
+        if not old_row:
+            logger.warning('XMLID %s not found when renaming to %s', old, new)
```

Comment:
I don't know if it's a good idea to log this.

Reply:
Could help to detect typos

Reply:
I prefer this to be `info`, as there are legit cases where the XML-ID won't be found, and having it as warning may disturb. Example (real that happened on v12, but names may not be accurate): you have the XML-ID `sale.group_discount` that is now in v12 `account.group_discount`. On pre-migration of account, you do such renaming for having things in the proper place, but if you don't have `sale` account, the XML-ID won't exist.